composer/utils/inference.py

```
@@ -0,0 +1,175 @@
+    model = copy.deepcopy(model)
+
+    # Apply surgery algorithms in the given order
+    for alg in ensure_tuple(surgery_algs):
+        model = alg(model)
+
+    if load_path is not None:
+        # download checkpoint and load weights only
+        log.debug('Loading checkpoint at %s', load_path)
+        with tempfile.TemporaryDirectory() as tempdir:
+            composer_states_filepath, _, _ = download_checkpoint(path=load_path,
+                                                                 node_checkpoint_folder=tempdir,
+                                                                 object_store=load_object_store,
+                                                                 progress_bar=True)
+            state_dict = torch.load(composer_states_filepath, map_location='cpu')
+            missing_keys, unexpected_keys = model.load_state_dict(state_dict['state']['model'], strict=strict)
+            if len(missing_keys) > 0:
+                log.warning(f"Found these missing keys in the checkpoint: {', '.join(missing_keys)}")
+            if len(unexpected_keys) > 0:
+                log.warning(f"Found these unexpected keys in the checkpoint: {', '.join(unexpected_keys)}")
+
+    model.eval()
+    # Apply transformations (i.e., inference optimizations) in the given order
+    for transform in ensure_tuple(transforms):
+        model = transform(model)
+
+    is_remote_store = store_object_store is not None
+    tempdir_ctx = tempfile.TemporaryDirectory() if is_remote_store else contextlib.nullcontext(None)
+    with tempdir_ctx as tempdir:
+        if is_remote_store:
+            local_store_path = os.path.join(str(tempdir), 'model.export')
+        else:
+            local_store_path = store_path
+
+        if target_format == ExportFormat.TORCHSCRIPT:
+            export_model = None
+            try:
+                export_model = torch.jit.script(model)
+            except Exception as e:
```

Can we let the exception propagate through?

---

What do you mean? If script fails, we have to try tracing so need to catch exception.

---

Do you mean to rethrow if tracing also fails?

---

Ah I see, I think I missed the indentation. This logic works, though thoughts about doing something like this?

```python
if sampel_input is None:
    model = torch.jit.script(model)
else:
    model = torch.jit.trace(model, sample_input)
```

My slight preference is for the above (I think it's slightly easier to follow / reason though), but also understand if torch jit script is superior and we should use that whenever possible.